.devcontainer/Dockerfile

```diff
@@ -53,6 +76,12 @@ RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/nul
     apt-get update -qq && export DEBIAN_FRONTEND=noninteractive && \
     apt-get install -y --no-install-recommends cmake cmake-curses-gui
 
+# Install optional dependecies
+RUN apt-get update -qq && export DEBIAN_FRONTEND=noninteractive && \
+    apt-get install -y --no-install-recommends \
+        doxygen ccache cppcheck \
+        neovim emacs nano
```

Comment:
I appreciate the effort to split packages into 'necessary' vs 'optional', but it looks like you lost the `graphviz` packages in this change. `graphviz` is important for `doxygen`; we need that.

Reply:
oups nice catch!